client/src/components/SearchHighlighter.tsx

```
@@ -0,0 +1,531 @@
+    // DOM Manipulation to highlight terms using TreeWalker
+    useEffect(() => {
+        if (!isContentReady || !contentContainerRef.current || normalizedTerms.length === 0) {
+            setMatches({});
+            setTerms([]);
+            setCoOccurrenceCount(0);
+            return;
+        }
+
+        const container = contentContainerRef.current;
+        const newMatches: Record<string, MatchInstance[]> = {};
+        normalizedTerms.forEach(t => newMatches[t] = []);
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)